libraries/primitive_ensure_choregraphie.rb

```diff
@@ -0,0 +1,29 @@
+require_relative 'primitive_check_file'
+
+module Choregraphie
+  class EnsureChoregraphie < CheckFile
+    def initialize(file_path = ::File.join(::Chef::Config['file_cache_path'], 'unlock_choregraphie'), options = {})
+      super(file_path, options)
+    end
+
+
+    def register(choregraphie)
+      # We clear the resources list since we do not want other choregraphies
+      # with to think that they are protected.
+      choregraphie.resources.clear
+
+      choregraphie.before do |resource_name|
+        raise 'ensure_choregraphie primitive cannot be used on other events than resource convergence!' unless resource_name
+
+        if ::Choregraphie::Choregraphie.all.any? { |c| c.resources.include?(resource_name) }
+          ::File.write(@file_path, '')
```

Comment:
OPT: put a text inside the file to explain why. It will help to debug by whom this file is created

Reply:
done